listenbrainz/webserver/views/stats_api.py

```diff
@@ -409,6 +409,69 @@ def get_listening_activity(user_name: str):
     }})
 
 
+@stats_api_bp.get("/user/<user_name>/artist-activity")
+@crossdomain
+@ratelimit()
+def get_artist_activity(user_name: str):
+    """
+    Get the artist activity for user ``user_name``. The artist activity shows the total number of listens
+    for each artist along with their albums and corresponding listen counts.
+
+    A sample response from the endpoint may look like:
+
+    .. code-block:: json
+
+        {
+            "result": [
+                {
+                    "name": "Radiohead",
+                    "listen_count": 120,
+                    "albums": [
+                        {"name": "OK Computer", "listen_count": 45},
+                        {"name": "In Rainbows", "listen_count": 75}
+                    ]
+                },
+                {
+                    "name": "The Beatles",
+                    "listen_count": 95,
+                    "albums": [
+                        {"name": "Abbey Road", "listen_count": 60},
+                        {"name": "Revolver", "listen_count": 35}
+                    ]
+                }
+            ]
+        }
+
+    .. note::
+
+        - The example above shows artist activity data with two artists and their respective albums.
+        - The statistics are aggregated based on the number of listens recorded for each artist and their albums.
+
+    :statuscode 200: Successful query, you have data!
+    :statuscode 204: Statistics for the user haven't been calculated, empty response will be returned
+    :statuscode 400: Bad request, check ``response['error']`` for more details
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    """
+    data = _get_entity_stats(user_name, "release_groups", "total_release_group_count")
+    result = {}
+    for release_group in data["payload"]["release_groups"]:
+        artist_name = release_group["artist_name"]
```

Comment:
I deployed your PR on test.LB, I got an error saying 
```
  File "/code/listenbrainz/listenbrainz/webserver/views/stats_api.py", line 458, in get_artist_activity
    for release_group in data["payload"]["release_groups"]:
                         ~~~~^^^^^^^^^^^
TypeError: 'Response' object is not subscriptable
```
The problem is that `_get_entity_stats` returns a Flask Response object, but you're trying to access it like a dictionary. Instead you should fetch the stats directly from the DB here, and process it further.